lib/api/websiteHead.js

```diff
@@ -72,29 +71,31 @@ function websiteHead(request, log, callback) {
         // handle redirect all
         if (websiteConfig.getRedirectAllRequestsTo()) {
             return callback(null, corsHeaders,
-                websiteConfig.getRedirectAllRequestsTo(), objectKey);
+                websiteConfig.getRedirectAllRequestsTo(), reqObjectKey);
         }
 
         // find index document if "directory" sent in request
         if (reqObjectKey.endsWith('/')) {
-            objectKey += websiteConfig.getIndexDocument();
+            // eslint-disable-next-line no-param-reassign
+            request.objectKey += websiteConfig.getIndexDocument();
```

Comment:
I think it could be worth turning this small preprocessing section to add the website index into a helper function, especially since it is shared between `websiteHead` and `websiteGet`. For example a local `appendWebsiteIndexDocument(request)` could work, making it clear we're modifying the request object.

I also see that a lot of code is duplicated between HEAD and GET handling, obviously not related to this PR, but IMHO could be worth opening a cleanup ticket to take care of this at another occasion and reduce tech debt.

Reply:
I've moved the processing to an helper function.
I've also added some functionnal tests for the bugfix.
I've also noticed there are some code duplication in functional tests between HEAD and GET